### DISCUSSION TITLE
Support for "Custom fields" meta box?

### DISCUSSION THREAD

**Discussion by mattsherman** [woocommerce/woocommerce#42710] [WooCommerce New Product Editor]
Recently in the community Slack, a question was asked about support for the "Custom fields" meta box that is available in the classic product editor experience.

<img width="1095" alt="Screenshot 2023-12-12 at 11 01 08" src="https://github.com/woocommerce/woocommerce/assets/2098816/8435b988-294d-4c6e-8967-06791eee88c7">

<img width="773" alt="Screenshot 2023-12-12 at 10 55 21" src="https://github.com/woocommerce/woocommerce/assets/2098816/1978013a-155b-4ee1-b309-97ad02da5373">

This allows users to access any custom fields not prefixed with `_` without having to have a custom UI for them.

**In order to provide similar support in the new product editor, I propose that we implement and provide a similar component that can be added to the product editor layout templates (via the API hooks). I believe that a solution like this will provide a "bridge" that will help more merchants to be able to switch to the new product editor.**

Ideally, we'd be able to detect if the "custom fields" meta box was displayed in the classic experience and auto-add it if it was. If for whatever reason we are not able to detect this, we could provide a toggle to show/hide this. By default, it would be hidden.

What do others think about this? I'm interested in everybody's opinions... extension developers, merchants, Woo devs/designers!

**Comment by NTShop** [comment]
This is definitely needed, many sites use the Custom Fields metabox for a variety of reasons specific to their business. For example, adding new meta data, tracking the values of existing meta data, etc.

**Comment by pmcpinto** [comment]
> In order to provide similar support in the new product editor, I propose that we implement and provide a similar component that can be added to the product editor layout templates (via the API hooks). I believe that a solution like this will provide a "bridge" that will help more merchants to be able to switch to the new product editor.
> Ideally, we'd be able to detect if the "custom fields" meta box was displayed in the classic experience and auto-add it if it was. If for whatever reason we are not able to detect this, we could provide a toggle to show/hide this. By default, it would be hidden.

This solution makes sense. We can leverage the sidebar we plan to implement to surface the toggle. I wonder if the General tab is the best place to surface this feature. cc @j111q for feedback.

**Comment by louwie17** [comment]
I would almost advocate for its own tab, given in the classic experience it also gets its own metabox.
We could show it by default for now, although a setting to hide/view it would be handy, but this is not something we currently have in our designs yet.
We could add a setting in the **WooCommerce > Settings > Products** menu ( maybe under a new section for the new product editor ), that only shows up when the feature flag is enabled.

**Reply by mattsherman** [reply]
> We could show it by default for now, although a setting to hide/view it would be handy, but this is not something we currently have in our designs yet.

It is not shown by default in the classic experience.

> We could add a setting in the WooCommerce > Settings > Products menu ( maybe under a new section for the new product editor ), that only shows up when the feature flag is enabled.

I think we should explore putting the toggle in the more (...) menu in the new product editor.

**Reply by NTShop** [reply]
In my WC 8.x install the custom fields metabox is shown by default. But maybe it is due to a cookie issue in relation to sub-folder installation. Not sure.

**Comment by j111q** [comment]
I created an issue for this here: #44169. Thank you for flagging it @mattsherman!
